packages/web3torrent/src/clients/web3t-channel-client.ts

```diff
@@ -146,6 +252,9 @@ export class Web3TorrentChannelClient {
 
   async pushMessage(message: Message<ChannelResult>) {
     await this.channelClient.pushMessage(message);
+    const channelResult: ChannelResult = message.data;
+    const channelId = channelResult.channelId;
+    this.openChannels = {...this.openChannels, [channelId]: convertToChannelState(channelResult)};
```

Comment:
Seems kind of odd to me. Wouldn't `pushMessage` need to cause the wallet to evaluate the message and then emit `ChannelUpdated`?

Reply:
Yes that is what happens: but the currently the class does not update the `openChannels`(aka `channelCache` when `ChannelUpdated` is emitted from the wallet.

The pattern that I will stick to (for now) is to update the cache with whatever the wallet _returns_, rather than with something it emits. It seemed weird to try and bind to those events inside this class.